src/syncerd/bitcoin_syncer.rs

```diff
@@ -411,6 +413,118 @@ fn query_addr_history(
     Ok(addr_txs)
 }
 
+/// Returns the script code used for spending a P2WPKH output if this script is a script pubkey
+/// for a P2WPKH output. The `scriptCode` is described in [BIP143].
+///
+/// [BIP143]: <https://github.com/bitcoin/bips/blob/99701f68a88ce33b2d0838eb84e115cef505b4c2/bip-0143.mediawiki>
+/// FIXME: switch to impl in https://github.com/rust-bitcoin/rust-bitcoin/commit/d882b68a2cbba8e643695552f38e4b36558f9617 once released
+fn p2wpkh_script_code(script: &bitcoin::Script) -> Script {
+    bitcoin::blockdata::script::Builder::new()
+        .push_opcode(bitcoin::blockdata::opcodes::all::OP_DUP)
+        .push_opcode(bitcoin::blockdata::opcodes::all::OP_HASH160)
+        .push_slice(&script[2..])
+        .push_opcode(bitcoin::blockdata::opcodes::all::OP_EQUALVERIFY)
+        .push_opcode(bitcoin::blockdata::opcodes::all::OP_CHECKSIG)
+        .into_script()
+}
+
+fn sweep_address(
+    source_private_key: [u8; 32],
+    source_address: bitcoin::Address,
+    dest_address: bitcoin::Address,
+    electrum_server: String,
+    network: bitcoin::Network,
+) -> Result<Vec<Vec<u8>>, Error> {
+    match source_address.address_type() {
+        Some(bitcoin::AddressType::P2wpkh) => {}
+        Some(address_type) => {
+            return Err(Error::Farcaster(format!(
+                "Sweeping addresses only supports native segwit v1 addresses. Address has type: {}",
+                address_type
+            )));
+        }
+        None => return Err(Error::Farcaster(format!("Invalid to be swept address"))),
+    }
+
+    let sk = bitcoin::PrivateKey::from_slice(&source_private_key, network)?;
+    let pk = bitcoin::PublicKey::from_private_key(bitcoin::secp256k1::SECP256K1, &sk);
+
+    let client = Client::new(&electrum_server)?;
+    let unspent_txs = client.script_list_unspent(&source_address.script_pubkey())?;
+
+    if unspent_txs.len() == 0 {
+        return Ok(vec![]);
+    }
+
+    let in_amount = unspent_txs
+        .iter()
+        .fold(0, |acc, unspent_tx| acc + unspent_tx.value);
+    let inputs: Vec<bitcoin::TxIn> = unspent_txs
+        .iter()
+        .map(|unspent_output| bitcoin::TxIn {
+            previous_output: bitcoin::OutPoint {
+                txid: unspent_output.tx_hash,
+                vout: unspent_output.tx_pos as u32,
+            },
+            script_sig: bitcoin::Script::default(),
+            sequence: (1 << 31) as u32,
+            witness: bitcoin::Witness::new(),
+        })
+        .collect();
+
+    let mut unsigned_tx = bitcoin::Transaction {
+        version: 2,
+        lock_time: 0,
+        input: inputs,
+        output: vec![bitcoin::TxOut {
+            value: in_amount,
+            script_pubkey: dest_address.script_pubkey(),
+        }],
+    };
+
+    let blocks_until_confirmation = 2;
+    let fee_btc_per_kvb = client.estimate_fee(blocks_until_confirmation)?;
+    let fee_sat_per_vb = fee_btc_per_kvb * 1e5;
+    let vsize_per_input_witness = 28;
```

Comment:
document, plox

Reply:
Done in https://github.com/farcaster-project/farcaster-node/pull/499/commits/e0e7aa9bb7f17ee1a314900ade88589e6ded3097

Reply:
Misplaced a parenthesis -_- fixed in https://github.com/farcaster-project/farcaster-node/pull/499/commits/62fe50216ca1586a83bb6720e6148579282cf389